packages/typescript/src/context/ts-context.ts

```diff
@@ -0,0 +1,16 @@
+import { ComponentContext, createContext, useContext } from "@alloy-js/core";
+
+export interface TypeScriptContext {
+  /** If in a type context */
+  type?: boolean;
+}
+
+/**
+ * Provides scopes for instance and static private members.
+ */
+export const TypeScriptContext: ComponentContext<TypeScriptContext> =
```

Comment:
Not really sure about the name for this or even if we should reuse something like scope(Didn't felt right)

Reply:
I don't think we want anything generalized here. I'd do `TypeRefContext: ComponentContext<TypeRefContext>` maybe?

I would also add `isTypeRefContext()` accessor.